vm/systemSmartContracts/eei.go

```diff
@@ -272,6 +280,63 @@ func (host *vmContext) Transfer(
 	return nil
 }
 
+// ProcessBuiltInFunction will execute process if sender and destination is same shard/sovereign
+func (host *vmContext) ProcessBuiltInFunction(
+	destination []byte,
+	sender []byte,
+	value *big.Int,
+	input []byte,
+	gasLimit uint64,
+) error {
+	isIntraBuiltIn, vmInput := host.isIntraShardBuiltInCall(destination, sender, value, input, gasLimit)
```

Comment:
is this check needed? couldn't the `vmOutput, err := host.blockChainHook.ProcessBuiltInFunction(vmInput)` do this internally? 

could this open the execution of BuiltInFunctions on the metachain, because it could be "intraShardBuiltInCall"?

Reply:
I dod not want to add inside processBuiltIn - as it would execute at least the sender part, which is wrong in this case for mainnet.

The check is done especially in sense to work on sovereign chains. In case of mainnet there is no way for both sender and destination to be in the metacain.